cmd/initialization/initialization.go

```diff
@@ -0,0 +1,207 @@
+package initialization
+
+import (
+	"os"
+
+	"github.com/alicenet/alicenet/config"
+	"github.com/alicenet/alicenet/logging"
+	toml "github.com/pelletier/go-toml"
+	"github.com/spf13/cobra"
+)
+
+// Command is the cobra.Command specifically for initialising the alicenet client.
+var Command = cobra.Command{
+	Use:   "init",
+	Short: "Initialize the files/folders required for running the alicenet client",
+	Long:  "Initialize the files/folders required for running the alicenet client",
+	Run:   initialiseFilesAndFolders,
+}
+
+func initialiseFilesAndFolders(cmd *cobra.Command, args []string) {
+	logger := logging.GetLogger("init").WithField("Component", cmd.Use)
+
+	path := config.Configuration.Initialization.Path
+	network := config.Configuration.Initialization.Network
+
+	if path == "" {
+		logger.Info("No path specified - defaulting to home directory")
+		path = os.Getenv("HOME")
+	}
+
+	if network == "" {
+		logger.Info("No environment specified - defaulting to mainnet")
+		network = "mainnet"
+	}
+
+	logger.Info("Initialising AliceNet configuration files and folders...")
+
+	// alicenet related paths and files
+	alicenetPath := path + "/alicenet"
+	envPath := alicenetPath + "/" + network
+	stateDBPath := envPath + "/stateDB"
+	transactionDBPath := envPath + "/transactionDB"
+	monitorDBPath := envPath + "/monitorDB"
+	keystoresPath := envPath + "/keystores"
+	keysPath := keystoresPath + "/keys"
+	configPath := envPath + "/config.toml"
+
+	paths := []string{alicenetPath, envPath, stateDBPath, transactionDBPath, monitorDBPath, keystoresPath, keysPath}
+
+	// loop through the paths checking to see if they exist and exit if any of them do
+	for _, path := range paths {
+		if _, err := os.Stat(path); !os.IsNotExist(err) {
+			logger.WithError(err).Fatal("Path already exists", path)
+			logger.Fatal("Remove all existing paths and try again")
+		}
+	}
+
+	// check for the existence of the config file and exit if it exists
+	if _, err := os.Stat(configPath); !os.IsNotExist(err) {
+		logger.WithError(err).Error("Config file already exists: ", configPath)
+		logger.Fatal("Remove the existing config file and try again")
+	}
+
+	// create the paths
+	for _, path := range paths {
+		err := os.MkdirAll(path, 0o755)
+		if err != nil {
+			logger.WithError(err).Fatal("Failed to create path: ", path)
+			// remove the root folder and its subfolders then exit
+			err := os.RemoveAll(alicenetPath)
+			if err != nil {
+				logger.WithError(err).Fatal("Failed to remove path: ", alicenetPath)
+				logger.Fatal("Please remove the root folder and its subfolders manually")
+			}
+			logger.Fatal("Exiting")
+		}
+	}
+
+	type LogLevel struct {
```

Comment:
Instead of re-creating the structs down below, can we not just annotate the structs defined in `./config/settings.go` ? We can use the `tags` omit-empty I believe, we don't have comments for certain fields.